src/File.hs

```diff
@@ -56,3 +57,23 @@ replaceIO find replacement file =
     $ embedToFinal
     $ runIO
     $ (replace find replacement file)
+
+dropLine ::
+  Member File r =>
+  Text ->
+  FilePath ->
+  Sem r Bool
+dropLine find file = do
+  contents <- File.read file
+  let newContents = T.unlines $ filter (not . T.isInfixOf find) (T.lines contents)
+  when (contents /= newContents) $ do
+    File.write file newContents
+  return $ contents /= newContents
```

Comment:
This looks fragile.   Can we do better?  Maybe ask `nix eval` for line info, or implement it as `dropAttrLine` that deletes `^\s*$attr.*=.*;`?  Also could use some test cases, which we're trying to improve coverage of.

Everything you had up until the `deleteVendor` stuff looks good; if you rebase to just have that I can merge those portions and we can handle this in another PR (up to you).